pkg/kubelet/pleg/generic.go

```diff
@@ -431,6 +433,24 @@ func getContainerState(pod *kubecontainer.Pod, cid *kubecontainer.ContainerID) p
 	return state
 }
 
+func updateRunningPodAndContainerMetrics(pods []*kubecontainer.Pod) {
+	// Set the number of running pods in the parameter
+	metrics.RunningPodCount.Set(float64(len(pods)))
+	// intermediate map to store the count of each "container_state"
+	containerStateCount := make(map[string]int)
+
+	for _, pod := range pods {
+		containers := pod.Containers
```

Comment:
Should InitContainers be included ?

Reply:
Makes sense introducing it in a new GaugeVec(`running_initcontainer_count` something like this maybe)
@logicalhan what do you think?

Reply:
I found that container.Pod doesn't expose initcontainers.
Edited comment above for Sandboxes.

If this makes sense, I can send out a PR.

Reply:
We could add this, although i'm not 100% sure what the value of it would be.  It should be exactly equal to running pod count except in rare scenarios when the kubelet is recreating a sandbox.  But i'm not entirely sure why knowing that a sandbox is being recreated is useful... What did you intend to use it for?

Reply:
Assuming the sandbox count settles down after the recreation, probably we don't need the extra gauge.